File: go_backend/models/objeto.go
package models

type Objeto struct {
	ID            string   `json:"id"`
	CajaID        string   `json:"caja_id"`
	UsuarioID     string   `json:"usuario_id"`
	Nombre        string   `json:"nombre"`
	NombreNorm    string   `json:"-"`
	Descripcion   string   `json:"descripcion,omitempty"`
	Cantidad      int      `json:"cantidad"`
	EsInsumo      bool     `json:"es_insumo"`
	ValorEstimado *float64 `json:"valor_estimado,omitempty"`
	CreatedAt     Time     `json:"created_at"`
	UpdatedAt     Time     `json:"updated_at"`
}

File: flutter_app/lib/core/models/objeto.dart
class Objeto {
  final String id;
  final String cajaId;
  final String nombre;
  final String descripcion;
  final int cantidad;
  final bool esInsumo;
  final double? valor;
  final bool tieneFotos;

  Objeto({
    required this.id,
    required this.cajaId,
    required this.nombre,
    this.descripcion = '',
    this.cantidad = 1,
    this.esInsumo = false,
    this.valor,
    this.tieneFotos = false,
  });

  factory Objeto.fromJson(Map<String, dynamic> json) => Objeto(
        id: json['id'] as String,
        cajaId: json['caja_id'] as String,
        nombre: json['nombre'] as String,
        descripcion: json['descripcion'] as String? ?? '',
        cantidad: json['cantidad'] as int? ?? 1,
        esInsumo: json['es_insumo'] as bool? ?? false,
        valor: json['valor'] as double?,
        tieneFotos: json['tiene_fotos'] as bool? ?? false,
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'nombre': nombre,
      };
}
